cvmfs/sqlitemem.h

```diff
@@ -0,0 +1,303 @@
+/**
+ * This file is part of the CernVM File System.
+ */
+
+#ifndef CVMFS_SQLITEMEM_H_
+#define CVMFS_SQLITEMEM_H_
+
+#include <inttypes.h>
+#include <pthread.h>
+#include <stdint.h>
+
+#include <cstdlib>
+#include <vector>
+
+#include "duplex_sqlite3.h"
+#include "gtest/gtest_prod.h"
+
+namespace sqlite {
+
+/**
+ * The MemoryManager uses the sqlite hooks to optimize memory allocations.  It
+ * is tuned for reading the cvmfs file catalogs.  It provides a page cache of
+ * a few MB, a small scratch space and per-database lookaside buffers.  It also
+ * contains a "general purpose" malloc/free implementation tailored to the
+ * behavior of sqlite memory allocation.
+ *
+ * It is implemented as a singleton.  GetInstance() will reserve memory blocks,
+ * AssignGlobalArenas will set the global sqlite configuration and
+ * CleanupInstance frees the memory blocks.  These three routines are not
+ * thread-safe.  AssignGlobalArenas has to be called as first sqlite operation,
+ * CleanupInstance has to be called after sqlite3_shutdown.  If the singleton
+ * is alive, read-only sqlite databases will automatically use it for their
+ * lookaside buffers (see sql.h).  Assignment of lookaside buffers is
+ * thread-safe.
+ */
+class MemoryManager {
```

Comment:
`MemoryManager` implies that this is a general purpose manager while it is actually tailored for *sqlite*. `SqliteMemoryManager` maybe?